src/pydvl/value/least_core/_common.py

```diff
@@ -1,108 +1,167 @@
-        defining the minimum and maximum values of that decision variable. Use
-        ``None`` to indicate that there is no bound. By default, bounds are
-        ``(0, None)`` (all decision variables are non-negative). If a single
-        tuple ``(min, max)`` is provided, then ``min`` and ``max`` will serve as
-        bounds for all decision variables.
-    :param options: A dictionary of solver options. Refer to scipy's
-        documentation for all possible values.
+        & x in \mathcal{R}^n , \\
+        & e \ge 0
+
+     where :math:`x` is a vector of decision variables; ,
+    :math:`b_{ub}`, :math:`b_{eq}`, :math:`l`, and :math:`u` are vectors; and
+    :math:`A_{ub}` and :math:`A_{eq}` are matrices.
+
+    :param A_eq: The equality constraint matrix. Each row of ``A_eq`` specifies the
+        coefficients of a linear equality constraint on ``x``.
+    :param b_eq: The equality constraint vector. Each element of ``A_eq @ x`` must equal
+        the corresponding element of ``b_eq``.
+    :param A_lb: The inequality constraint matrix. Each row of ``A_lb`` specifies the
+        coefficients of a linear inequality constraint on ``x``.
+    :param b_lb: The inequality constraint vector. Each element represents a
+        lower bound on the corresponding value of ``A_lb @ x``.
+    :param options: Keyword arguments that will be used to select a solver
+        and to configure it. Refer to the following page for all possible options:
+        https://www.cvxpy.org/tutorial/advanced/index.html#setting-solver-options
     """
-    logger.debug(
-        f"Solving linear programming problem: {c=}, {A_eq=}, {b_eq=}, {A_ub=}, {b_ub=}"
-    )
-
-    result: scipy.optimize.OptimizeResult = scipy.optimize.linprog(
-        c,
-        A_ub=A_ub,
-        b_ub=b_ub,
-        A_eq=A_eq,
-        b_eq=b_eq,
-        bounds=bounds,
-        method="highs-ipm",
-        options=options,
-    )
+    logger.debug(f"Solving linear program : {A_eq=}, {b_eq=}, {A_lb=}, {b_lb=}")
+
+    x = cp.Variable(n_variables)
+    e = cp.Variable()
+    objective = cp.Minimize(e)
+    constraints = [
+        e >= 0,
+        A_eq @ x == b_eq,
+        A_lb @ x + e * np.ones(len(A_lb)) >= b_lb,
+    ]
+    problem = cp.Problem(objective, constraints)
+
+    solver = options.pop("solver", cp.ECOS)
+
+    try:
+        problem.solve(solver=solver, **options)
+    except cp.error.SolverError as err:
+        raise ValueError("Could not solve linear program") from err
+
+    if problem.status in cp.settings.SOLUTION_PRESENT:
+        logger.debug("Problem was solved")
+        if problem.status in [cp.settings.OPTIMAL_INACCURATE, cp.settings.USER_LIMIT]:
+            warnings.warn(
+                "Solver terminated early. Consider increasing the solver's "
+                "maximum number of iterations in options",
+                RuntimeWarning,
+            )
+        least_core_value = e.value.item()
```

Comment:
See my comment below, value is already quite overloaded and the paper uses "subsidy"
```suggestion
        subsidy = e.value.item()
```

Reply:
You're right, it's overloaded. The paper also uses the term "deficit" and comparing the definition of both terms in a dictionary I would lean towards using "deficit" rather than subsidy.

* **deficit**: the amount by which something, especially a sum of money, is too small.
* **subsidy**: a sum of money granted to support an undertaking held to be in the public interest.

Reply:
I just double-checked and it seems to be also just called **least core value** in the literature: https://www.sciencedirect.com/science/article/pii/S1572528613000042#br000100

Reply:
Still, I'd avoid the overloading. For us values are assigned to samples.

Reply:
Hmmm, I see. your reasoning makes sense. I'll change it to subsidy.